heartbeat/scripts/mage/package.go

```diff
@@ -65,5 +65,9 @@ func dashboards() error {
 
 // Fields generates a fields.yml for the Beat.
 func Fields() error {
-	return devtools.GenerateFieldsYAML(devtools.OSSBeatDir("monitors/active"))
+	err := devtools.GenerateFieldsYAML(devtools.OSSBeatDir())
+	if err != nil {
+		return err
+	}
+	return devtools.GenerateAllInOneFieldsGo()
```

Comment:
When this is run from x-pack/heartbeat it will create x-pack/heartbeat/include/fields.go, but since there are no new fields being introduced we can skip this part. We'd only want to introduce an x-pack/heartbeat/include/fields.go file if X-Pack heartbeat added new fields. And if this file was added we'd only want it to include the new fields (not any of the libbeat or OSS heartbeat fields) since they are already included on the OSS side an inherited by x-pack.

You could use some license based conditionals like in Winlogbeat

https://github.com/elastic/beats/blob/c2272c192f2a19c4b3412dfea21e25c1ca180934/winlogbeat/scripts/mage/fields.go

or do the same work in both OSS and X-Pack but write the output to OSS only

`devtools.GenerateFieldsGo("fields.yml", devtools.OSSBeatDir("include/fields.go"))`

Reply:
It will generate new fields when the synthetics stuff is merged in #21436 , and in fact this PR is a bit of a blocker for that one. That said, if it's easier, I'd rather just include all the fields (including x-pack fields) in the OSS heartbeat and keep things simple. Does that work for you?

Reply:
I went with the latter approach you suggested. Thanks for the tips!